src/digitalIdentity-frontend/src/app/shared/pop-up/add-dito-credential-pop-up/add-dito-credential-pop-up.component.ts

```diff
@@ -0,0 +1,154 @@
+    private HttpService: BackendHttpService,
+    private router: Router,
+    private dialog_Ref: MatDialog,
+    @Inject(MAT_DIALOG_DATA) private data: { id: string, schemaId:string, alias:string }
+  ) {
+    if (isDevMode()) {
+      this.cancelButtonString = 'Ney!';
+    }
+    this.id = data.id;
+    this.schemaId = data.schemaId;
+    this.alias = data.alias;
+
+    this.attributeFormGroup = this.fb.group({
+      connection: [null, Validators.required],
+      attributes: [null]
+    })
+  }
+
+  async ngOnInit() {
+    this.schema = await this.getSchemaByID(this.schemaId);
+    this.getDI();
+    this.schema.attributes.forEach((attribute: any)=>{
+      let objAttribute:any = {};
+      objAttribute.name = attribute;
+      objAttribute.value = '';
+      this.attributesData.push(objAttribute);
+
+    });
+  }
+
+  getDI() {
+    const params = new HttpParams().append('authorization', 'passing');
+    this.HttpService.getRequest("Get all connection","/connection/all",params)
+      .then(
+        response => {
+          if (response.ok) {
+            this.DIData = response.body;
+            console.log(this.DIData)
+          }
+        }
+      )
+      .catch(response => {console.log("error"); console.log(response)})
+  }
+
+  getSchema() {
+    const params = new HttpParams().append('authorization', 'passing');
+    this.HttpService.getRequest("Get all schemas","/schema/all",params)
+      .then(
+        response => {
+          if (response.ok) {
+            this.schemaData = response.body
+            this.filteredSchemas = this.schemaData.slice();
+          }
+        }
+      )
+      .catch(response => {console.log("error"); console.log(response)})
+  }
+
+  async getSchemaByID(schemaID: string) {
+    const params = new HttpParams().append('authorization', 'passing');
+    let response = await this.HttpService.getRequest("Get all schemas","/schema/all",params)
+    if (response.ok) {
+      this.schemaData = response.body
+      this.filteredSchemas = this.schemaData.slice();
+      for (let schema of this.filteredSchemas) {
+        if (schema.id == schemaID) {
+          console.log(schema);
+          return schema;
+        } else {
+          console.log("Schema with this ID not found");
+        }
+      }
+    }
+  }
+
+  getSelectedDI(event: any){
+    this.selectedId = event.value;
+    console.log(event);
+  }
+
+  cancelButtonEvent() {
+    this.dialogRef.close();
+  }
+  async save(){
+    let params = new HttpParams().append('authorization', 'passing');
+    params = params.append('connectionId', this.selectedId);
+    params = params.append('credentialDefinitionId', this.id );
+    let body = {connectionId:this.id,credentialDefinitionId:this.selectedId,attributes:this.attributesData};
+
+    let response = await this.HttpService.postRequest("Issue a credential to an existing connection","/credential/issue",JSON.stringify(this.attributesData),params)
+      .then((response) => {
+        if (!response.ok) {
```

Comment:
switch these cases: first if(response.ok) then the error case

Reply:
What is the advantage of having this case there first please?

Reply:
Conventions. It makes it easier to read if we write down similar things in a similar way.